src/main/java/business/ArticleCollection.java

```diff
@@ -0,0 +1,85 @@
+package business;
+
+import business.Helper.*;
+import business.News.*;
+import business.NewsSources.*;
+
+import java.util.*;
+import java.util.concurrent.*;
+
+// an interface for presentation layer to access scraped articles
+public class ArticleCollection {
+    // map articles with their category
+    public static final HashMap<String, Collection<Article>> articlesByCategories = new HashMap<>();
+
+    // get all news outlet css info
+    public static final HashMap<String, NewsOutlet> newsOutlets = GetNewsOutlets.newsOutlets;
+
+    // generate previews from articles scraped
+    public static List<Preview> getPreviewsByCategory(String category) {
+        // load articles if they haven't been loaded before
+        if (articlesByCategories.get(category) == null) {
+            try {
+                loadArticlesByCategory(category);
+            } catch (InterruptedException e) {
+                // TODO: display error message when articles cannot be load
+                e.printStackTrace();
+                return new ArrayList<>();
+            }
+        }
+        return createPreviewsByCategory(category);
+
+
+    }
+    private static List<Preview> createPreviewsByCategory(String category){
+        // loop through articles in the category and create a list of their previews
+        ArrayList<Preview> matchedPreviews = new ArrayList<>();
+        for (Article article: articlesByCategories.get(category)) {
+            matchedPreviews.add(article.getPreview());
+        }
+
+        // sort article by published time
+        Collections.sort(matchedPreviews);
+
+        return matchedPreviews;
+    }
+
+
+    // MULTI THREADING!
+    private static void loadArticlesByCategory(String category) throws InterruptedException{
+        List<Article> safeArticleList = Collections.synchronizedList(new ArrayList<>());
+        ExecutorService es = Executors.newCachedThreadPool();
+//        CovidInfo.loadInfo();
+        for (String newsOutlet: newsOutlets.keySet()){
```

Comment:
Why don't we use `newsOutlets.values()` instead of `newsOutlets.keySet()`?
Then in line 56, we can directly pass the newsOutlet

Anyways naming each item `newsOutlet` while looping through `keySet()` may cause confusion

Reply:
Resolved, thanks!